debezium-testing/debezium-testing-openshift/src/test/java/io/debezium/testing/openshift/assertions/PlainKafkaAssertions.java

```diff
@@ -0,0 +1,37 @@
+package io.debezium.testing.openshift.assertions;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import java.time.Duration;
+import java.time.temporal.ChronoUnit;
+import java.util.Collections;
+import java.util.Properties;
+import java.util.stream.StreamSupport;
+
+import org.apache.kafka.clients.consumer.Consumer;
+import org.apache.kafka.clients.consumer.ConsumerRecords;
+import org.apache.kafka.clients.consumer.KafkaConsumer;
+
+public class PlainKafkaAssertions implements KafkaAssertions {
+
+    private final Properties kafkaConsumerProps;
+
+    public PlainKafkaAssertions(Properties kafkaConsumerProps) {
+        this.kafkaConsumerProps = kafkaConsumerProps;
+    }
+
+    @Override
+    public Consumer<String, String> getConsumer() {
+        return new KafkaConsumer<>(kafkaConsumerProps);
+    }
+
+    public void assertRecordsContain(String topic, String content) {
+        try (Consumer<String, String> consumer = new KafkaConsumer<>(kafkaConsumerProps)) {
+            consumer.subscribe(Collections.singleton(topic));
+            consumer.seekToBeginning(consumer.assignment());
+            ConsumerRecords<String, String> records = consumer.poll(Duration.of(10, ChronoUnit.SECONDS));
+            long matchingCount = StreamSupport.stream(records.records(topic).spliterator(), false).filter(r -> r.value().contains(content)).count();
```

Comment:
The logic is very similar to avro version so maybe it should be extracted and really only the assert itself should be provided by the implementation

Reply:
Possibly but since Avro version basically does not do any assert at this moment I'm not quite certain how similar these two will end up. I will look into it once that is done